app/Platform/Services/UserAgentService.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services;
+
+class UserAgentService
+{
+    public array $cache = [];
+
+    public function decode(string $userAgent): array
+    {
+        if (array_key_exists($userAgent, $this->cache)) {
+            return $this->cache[$userAgent];
+        }
+
+        $data = $this->parseUserAgent($userAgent);
+        $this->cache[$userAgent] = $data;
+
+        return $data;
+    }
```

Comment:
This one line is logically equivalent, and due to `??=` the cache value is inserted if it isn't present:
```php
return $this->cache[$userAgent] ??= $this->parseUserAgent($userAgent);
```

Reply:
done